### DISCUSSION TITLE
`getDcpMessages` best practices

### DISCUSSION THREAD

**Discussion by ghost** [opendcs/opendcs#262] [Q&A] [answered]
My plan is to use just the `getDcpMessages` with a  `SearchCritFile` for retrieving messages for selected sites. The messages will then be processed by custom code in Python.

- For this use case, what are all the valid formats for netlist that can be used with `getDcpMessages`? By trial and error, I discovered that netlist can either be `XML` or a simple `CSV` file. Looks like in the case of `CSV`, each new line needs a separate address.
- Where should the netlists be located? Is it `OPENDCS/edit-db/netlist/`?
- Is it better to split all sites into multiple netlists so that even if there is a delay with one set of sites, it doesn't affect the other sites?

**Comment by agilmore2** [comment]
There are two netlist storage formats, the legacy netlists for LRGS that
can be used both as a DDS filter for incoming messages, as well as
referenced by DDS clients. Second, the OpenDCS routing specs can use
netlists stored as OPENDCS objects, either in XML (edit-db/netlist) or in
SQL in the NETWORKLIST and NETWORKLISTENTRY tables. These are presented in
the OpenDCS Decodes Editor GUI and can be referenced by routing specs.

The netlists are sent during DDS connection setup for the routing specs.
The union of all of them is used as a filter on the DDS connection
messages. There's no concept of extra delays here, just whenever the DCP
sends the transmission and the system forwards it.

As far as netlist management goes, it's a lump or split thing, and it's up
to you how you want to manage your DCP retrievals.

**Comment by ghost** [comment]
Thank you.

- What does the legacy netlist look like? Is it a simple list of addresses separated by newline?
- It doesn't sound like splitting addresses into multiple netlists is helpful in terms of avoiding or reducing delays. Is that correct?

**Reply by MikeNeilson** [reply, marked as answer]
Delays would likely depend on volume, If you're only processing a handful of stations you can keep it in one. On the other hand if you wanted to process the entirety of the DCS data I would make sense run things in parallel only processing a smaller amount in each process.
